app/src/androidTest/java/com/github/se/project/ui/lesson/AddLessonTest.kt

```diff
@@ -171,22 +160,78 @@ class AddLessonTest {
   }
 
   @Test
   fun testInitialState() {
-    composeTestRule.setContent {
-      AddLessonScreen(navigationActions, mockProfiles, mockLessons, onMapReadyChange = {})
-    }
+    composeTestRule.setContent { AddLessonScreen(navigationActions, mockProfiles, mockLessons, {}) }
     composeTestRule.onNodeWithText("Select Date").assertExists()
     composeTestRule.onNodeWithText("Select Time").assertExists()
   }
+
+  @Test
+  fun testInstantLessonDisplaying() {
+    composeTestRule.setContent { AddLessonScreen(navigationActions, mockProfiles, mockLessons, {}) }
+
+    // Set Instant Lesson
+    composeTestRule.waitUntil(15000) {
+      composeTestRule.onNodeWithTag("instantButton").isDisplayed()
+    }
+    composeTestRule.onNodeWithTag("instantButton").performClick()
+
+    // Check that the map, Date, and Time buttons are not displayed
+    composeTestRule.onNodeWithTag("mapButton").assertIsNotDisplayed()
+    composeTestRule.onNodeWithTag("DateButton").assertIsNotDisplayed()
+    composeTestRule.onNodeWithTag("TimeButton").assertIsNotDisplayed()
+
+    // Ch
+    composeTestRule.onNodeWithTag("titleField").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("DescriptionField").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("subjectButton").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("checkbox_ENGLISH").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("priceRangeSlider").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("confirmButton").assertIsDisplayed()
+  }
+  /*
+    @Test
+    fun testInstantLesson() {
+      composeTestRule.setContent { AddLessonScreen(navigationActions, mockProfiles, mockLessons, {}) }
+
+      // Set Instant Lesson
+      composeTestRule.waitUntil(15000) {
+        composeTestRule.onNodeWithTag("instantButton").isDisplayed()
+      }
+      composeTestRule.onNodeWithTag("instantButton").performClick()
+      // Set Title and Description
+      composeTestRule.onNodeWithTag("titleField").performTextInput("Math Lesson")
+      composeTestRule.onNodeWithTag("DescriptionField").performTextInput("This is a math lesson.")
+
+      // Set Subject and Language
+      composeTestRule.onNodeWithTag("subjectButton").performClick()
+      composeTestRule.onNodeWithTag("dropdown${Subject.AICC}").performClick()
+      composeTestRule.onNodeWithTag("checkbox_ENGLISH").performClick()
+
+      composeTestRule.onNodeWithTag("confirmButton").performClick()
+      verify(navigationActions).navigateTo(anyString())
+    }
+  */
+  @Test
+  fun testInstantInvalid() {
+    composeTestRule.setContent { AddLessonScreen(navigationActions, mockProfiles, mockLessons, {}) }
+
+    // Set Instant Lesson
+    composeTestRule.waitUntil(15000) {
+      composeTestRule.onNodeWithTag("instantButton").isDisplayed()
+    }
+    composeTestRule.onNodeWithTag("instantButton").performClick()
+
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+    verify(navigationActions, never()).navigateTo(anyString())
+  }
```

Comment:
These tests are also coming in a later PR, so they're not necessary here, but why not